contrib/datastore/file/gcp/gcp.go

```diff
@@ -85,27 +94,232 @@ func (g GcpAdapter) CreateFileShare(ctx context.Context, fs *file.CreateFileShar
 }
 
 func (g GcpAdapter) GetFileShare(ctx context.Context, fs *file.GetFileShareRequest) (*file.GetFileShareResponse, error) {
-	panic("implement me")
+	isCreateOpDone, err := g.isOperationDone(createOp)
+	if g.checkForOperation(ctx, isCreateOpDone, createOp, "create", err) != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	isUpdateOpDone, err := g.isOperationDone(updateOp)
+	if g.checkForOperation(ctx, isUpdateOpDone, updateOp, "update", err) != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	isDeleteOpDone, err := g.isOperationDone(deleteOp)
+	if g.checkForOperation(ctx, isDeleteOpDone, deleteOp, "delete", err) != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	instance, err := g.GetInstance(ctx, fs.Fileshare)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+	log.Debugf("Found Instance %+v", instance)
+
+	fileShare, err := g.ParseFileShare(instance)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	if instance.State == "READY" {
+		// GCP FileStore Instance exists & ready to use
+		fileShare.Status = "available"
+		return &file.GetFileShareResponse{
+			Fileshare: fileShare,
+		}, nil
+	} else {
+		// GCP FileStore Instance exists but is not ready to use
+		return &file.GetFileShareResponse{
+			Fileshare: fileShare,
+		}, fmt.Errorf("instance %s is in %s", instance.Name, instance.State)
+	}
 }
 
 func (g GcpAdapter) ListFileShare(ctx context.Context, fs *file.ListFileShareRequest) (*file.ListFileShareResponse, error) {
-	panic("implement me")
+	instances, err := g.instancesService.List(g.locationURI(g.backend.Region, "-")).Context(ctx).Do()
+	if err != nil {
+		log.Error(err)
```

Comment:
This will simply log the error sent by GCP, right?

Reply:
Yes, It will log the message in driver before returning to service layer